src/gui/qgscolorrampbutton.cpp

```diff
@@ -69,7 +69,8 @@ QSize QgsColorRampButton::sizeHint() const
 #ifdef Q_OS_WIN
   return QSize( 120, 22 );
 #else
-  return QSize( 120, 28 );
+  // Adjust height for HiDPI screens
+  return QSize( 120, std::max( static_cast<int>( fontMetrics().height( ) * 1.4 ), 28 ) );
```

Comment:
don't forget to multiply font metrics based sizes by UI_SCALE_FACTOR ( I realise that here it's not going to change anything since current UI_SCALE_FACTOR only varies on windows, but it's more future safe to multiply anyway in case that logic changes in future.